### DISCUSSION TITLE
It is illegal to charge for CoPilot.

### DISCUSSION THREAD

**Discussion by gitcnd** [community/community#30190] [Copilot Conversations]
_"GitHub [Copilot](https://github.com/features/copilot) is powered by Codex, a generative pretrained AI model created by OpenAI. It has been trained on natural language text and source code from publicly available sources, **including code in public repositories on GitHub.**"_

"public repositories on GitHub" are not just "free code" to train AI on, they're *licensed* and carry plenty of **PROHIBITIONS**, which contain assorted legal restrictions including **ATTRIBUTION**, source-code-availability, and in some cases terms about pricing restrictions, like ..."provided that you do not charge" ...

It's not just offensive, it's literally illegal that you exploit the work of myself and most other free and open source contributors, and then charge us for a derivative product you've built that's based on our FREE contribution - and doubly offensive that you single out assorted contributors to enjoy this product for free based on some kind of perverted popularity metric.  The people who deserve to use this product for free is everyone who contributed to it (every author of every product it was trained on), not just the lucky few whose work happened to become "popular".  

"Popularity" is a *blatantly* offensive metric.  It absolutely needs to be changed to "Contributor", not just for ethical reasons, but also because of legal ones.

**Comment by hexxone** [comment]
Yep. I am also NOT gonna Pay for my OWN public Source Code being taken, transformed and suggested back to me.
For companies or profit-oriented usages I could see it, but 10$/mo for EVERYONE in this day and age of subscriptions-hell is just unreasonable.
Anyway, this is really just a Microsoft thing to do.
Not like they gonna care for any community if there is $$$ in sight.

**Comment by Shybella** [comment]
It doesn't even have the code edit feature that OpenAI has. In my opinion, without code edit, it's useless. 

I cancelled it.

**Comment by CiberNin** [comment]
That's a bold hypothesis. Why not put your money where your mouth is an go bother some lawyers.

**Comment by MulleDK19** [comment]
It's not illegal. If you're uploading your code to GitHub, you agreed to the license terms, which clearly states you grant them the right to use your code to make or improve GitHub products.

**Comment by gitcnd** [comment]
@MulleDK19 - Here's the terms:
https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#d-user-generated-content
" and the rights you grant us are limited to those we need to provide the service. "
LIMITED means limited - and does NOT include "make or improve GitHub products" (which is not part of their terms by the way).
 
Also, show me the line where it releases them from their obligation to adhere to the terms of all our LICENSE files, to not obey attribution rules, and to bill us...

**Reply by MulleDK19** [reply]
It's _right_ there.. Section 4..

---

 **You grant us and our legal successors the right to** store, archive, **parse**, and display Your Content, and make incidental copies, **as necessary to provide the Service**,
### including improving the Service over time.

---

Which de facto includes making a new product, as that's part of improving the service. That's why it's called **GitHub** Copilot and not just Copilot, as calling it GitHub Copilot makes it part of the service, and thus allows them to use that clause.

Your license files don't mean shit when you've already agreed to grant them the right to do whatever they want with your code to improve the service.

You may not like it, but they're legally covered..

**Comment by soulomoon** [comment]
It split the community using politic strategic to grant the use privilege to thoese who can make louder voice.
While It should be free to all those code were used to train the product.
Oh, it does not profit if they do so.

**Comment by melonmouse** [comment]
Here's an interesting take on this issue: https://thanksforthecode.com/
The idea is to thank every single github repo owner. You can pass in a parameter so it does the thanking on behalf of a project: https://thanksforthecode.com/?name=PROJECT_NAME
Its meant to make people think about what attribution means in this context.

**Comment by DamianEstey** [comment]
I am going to give my perspective on the Copilot situation.
I think that when viewing this situation from how the artifical intelligence was trained, it does seem as though it should be illegal since the content it was trained on is community work.
However, if you look at it as though the artifical intelligence was actually trained on this work to then apply it as though it was a programmer itself; it doesn't seem illegal anymore.
We as human beings who know how to code likely learned a thing or two from GitHub when we learned how to code, does that mean our content should always be free?

**Reply by gitcnd** [reply]
To continue your analogy - is it OK for you to charge the people who trained you for free, when 100% of your knowledge is acquired solely from those people, and when you're a robot which does not need the money?

The reason we contribute to github **for free**, is because we're nice and generous people.  That's precisely the reason why co-pilot should never be charging **_*us*_** money.  To do so, is blatant and _repulsive_ exploitation.

**Comment by MasterInQuestion** [comment]
Complain to the court.

**Comment by andarbekbarsukov** [comment]
Using open-source code without proper licensing can lead to legal issues, which a [mortgage fraud lawyer](https://pearllemonlegal.com/mortgage-fraud-lawyer/) could explain.
